python/pyiceberg/typedef.py

```diff
@@ -155,3 +156,6 @@ def __eq__(self, other: Any) -> bool:
 
     def __repr__(self) -> str:
         return f"{self.__class__.__name__}[{', '.join(f'{key}={repr(value)}' for key, value in self.__dict__.items() if not key.startswith('_'))}]"
+    
+    def fields(self) -> List[str]:
+        return [self.__getattribute__(v) if hasattr(self, v) else None for v in self._position_to_field_name.values()]
```

Comment:
I couldn't find a more elegant way to return an array of fields preserving declaration order and preserving None for unset optional fields.